src/v8runner/src/v8runner.os

```diff
@@ -110,6 +110,110 @@
 	
 КонецПроцедуры
 
+// Выполнить объединение текущей конфигурации с файлом (с использованием файла настроек)
+//
+// Параметры:
+//   ФайлКонфигурации - Строка - имя cf-файла с объединяемой конфигурацией.
+//   ФайлНастроек - Строка - имя файла с настройками объединения конфигураций.
+//   ПоставитьНаПоддержку - Булево, Неопределено - флаг необходимости установки конфигурации на поддержку.
+//		Неопределено: по умолчанию флаг не передается.
+//		Истина: конфигурация ставится на поддержку, если есть такая возможность.
+//		Ложь: конфигурация не ставится на поддержку, даже если есть такая вожзможность.
+//   ВключитьВОбъединениеОбъектыПоНеразрешеннымСсылкам - Булево, Неопределено - флаг небходимости включения
+//		в объединение объектов, не включенных в список объединяемых и отсутствующих в основной конфигурации.
+//		Неопределено: по умолчанию флаг не передается.
+//		Истина: включать объекты с неразрешенными ссылками в список объединяемых.
+//		Ложь: очищать ссылки на объекты с неразрешенными ссылками.
+//   Принудительно - Булево - флаг принудительного обновления.
+//		Обновление будет производиться в случае наличия:
+//			предупреждений об удаляемых объектов, на которые есть ссылки в объектах, 
+//				не участвующих в объединении (такие объекты будут исключены из объединения);
+//			предупреждений о применении настроек.
+//		Если не указан, то объединение будет прервано в вышеуказанных случаях.
+//
+Процедура ОбъединитьКонфигурациюСФайлом(Знач ФайлКонфигурации, 
+								 Знач ФайлНастроек, 
+								 Знач ПоставитьНаПоддержку = Неопределено, 
+								 Знач ВключитьВОбъединениеОбъектыПоНеразрешеннымСсылкам = Неопределено, 
+								 Знач Принудительно = Ложь) Экспорт
+	
+	Параметры = СтандартныеПараметрыЗапускаКонфигуратора();
+	Параметры.Добавить("/MergeCfg " + ОбернутьВКавычки(ФайлКонфигурации));
+	Параметры.Добавить("-Settings " + ОбернутьВКавычки(ФайлНастроек));
+
+	Если ПоставитьНаПоддержку = Истина Тогда
+		Параметры.Добавить("-EnableSupport");
+	ИначеЕсли ПоставитьНаПоддержку = Ложь Тогда
+		Параметры.Добавить("-DisableSupport");
+	КонецЕсли;
+
+	Если ВключитьВОбъединениеОбъектыПоНеразрешеннымСсылкам = Истина Тогда
+		Параметры.Добавить("-IncludeObjectsByUnresolvedRefs");
+	ИначеЕсли ВключитьВОбъединениеОбъектыПоНеразрешеннымСсылкам = Ложь Тогда
+		Параметры.Добавить("-ClearUnresolvedRefs");
+	КонецЕсли;
+
+	Если Принудительно Тогда
+		Параметры.Добавить("-force");
+	КонецЕсли;
+
+	ВыполнитьКоманду(Параметры);
+
+КонецПроцедуры
+
+Процедура СравнитьКонфигурации(Знач ТипПервойКонфигурации,
```

Comment:
Класс так класс.
ОбъединитьКонфигурациюСФайлом хотя бы можно смержить?

22 июля 2016 г., 11:21 пользователь Artur Ayukhanov <
notifications@github.com> написал:

> In src/v8runner/src/v8runner.os
> <https://github.com/EvilBeaver/oscript-library/pull/132#discussion_r71843153>
> :
>
> > +
> > +	Если ВключитьВОбъединениеОбъектыПоНеразрешеннымСсылкам = Истина Тогда
> > +		Параметры.Добавить("-IncludeObjectsByUnresolvedRefs");
> > +	ИначеЕсли ВключитьВОбъединениеОбъектыПоНеразрешеннымСсылкам = Ложь Тогда
> > +		Параметры.Добавить("-ClearUnresolvedRefs");
> > +	КонецЕсли;
> > +
> > +	Если Принудительно Тогда
> > +		Параметры.Добавить("-force");
> > +	КонецЕсли;
> > +
> > +	ВыполнитьКоманду(Параметры);
> > +
> > +КонецПроцедуры
> > +
> > +Процедура СравнитьКонфигурации(Знач ТипПервойКонфигурации,
>
> может просто сделать два вида апи? Одно в command line стиле, как сейчас,
> а второе в объектном?
>
> 2 голоса за отдельный класс ( @EvilBeaver <https://github.com/EvilBeaver>
> @artbear <https://github.com/artbear> )
> @nixel2007 <https://github.com/nixel2007> @pumbaEO
> <https://github.com/pumbaEO> как вы проголосуете?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/EvilBeaver/oscript-library/pull/132/files/0f9d389a59ee6637fd1007ef8884b01fe95d498a#r71843153>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABFJKBqlQaA6F4qJ1D31k9mkSig88EhKks5qYH2XgaJpZM4JQioz>
> .
>